packages/itwin/grouping-mapping-widget/src/widget/components/viewerUtils.tsx

```diff
@@ -95,9 +95,9 @@ export const hideElementsByQuery = async (
   if (set.elements) {
     const ids = [...set.elements];
     hideElements(ids, replace);
-    return ids;
+    return { keySet, ids };
```

Comment:
Doesn't KeySet already contain ids? This looks like a duplication of data. If there is no required accessor in KeySet, maybe it is worth to contact the `@itwin/presentation-common` lib maintainers?

Reply:
It does, but it is not the complete set of ID's needed to render the element. That is where ```getHiliteSet``` comes into play. I cache the result of ```getHiliteSet``` even though it states that 
```
Note: The provider caches result of the last request, so subsequent requests
for the same input doesn't cost.
```
because it still makes a network request and there is a considerable slow down.